components/org.wso2.carbon.deployment.notifier/src/test/java/org/wso2/carbon/deployment/notifier/deployers/FaultyDeployer2.java

```diff
@@ -68,7 +78,11 @@ public void init() {
 
     public String deploy(Artifact artifact) throws CarbonDeploymentException {
         logger.info("Deploying : " + artifact.getName());
-        sample1Deployed = false;
+        if (artifact.getName().contains("sample1.txt")) {
```

Comment:
Since these are test artifacts if there is a NPE then the test will fail. So do we need to do a null check?Since these are test artifacts if there is a NPE then the test will fail. So do we need to do a null check?